src/main/java/igym/entities/User.java

```diff
@@ -13,24 +13,48 @@ public class User {
     private UUID id;
     private String name;
 
-    public User(UUID id, String name) {
-        this.id = id;
+    public User() {
+
+    }
+
+    public User(String name) {
         this.name = name;
     }
 
     public UUID getId() {
         return id;
     }
 
-    public void setId(UUID id) {
-        this.id = id;
-    }
-
     public String getName() {
         return name;
     }
 
     public void setName(String name) {
         this.name = name;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((id == null) ? 0 : id.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        User other = (User) obj;
+        if (id == null) {
+            if (other.id != null)
+                return false;
+        } else if (!id.equals(other.id))
+            return false;
+        return true;
+    }
```

Comment:
We must write a unit test for each condition of this method.

Reply:
```suggestion
public boolean equals(Object obj) {
    if (this == obj) return true;
    if (obj == null || getClass() != obj.getClass()) return false;
    User other = (User) obj;
    return id != null && id.equals(other.id);
}
```

I suggest this change to simplify the logic. 
Bellow I'm sharing with you the draft that I wrote to explain to myself the logic behind the last line of the refactor.

```
// id == null  ==> FALSE
// id == null && other.id != null ==> FALSE
// id == null && other.id == null ==> TRUE
// id != null && id != other.id ==> FALSE
// id != null && id.equals(other.id) ==> TRUE
```

Reply:
good suggestions! I will accept the changes and write the test for it